src/set_relation/isl_str_manipulation.cc

```diff
@@ -0,0 +1,291 @@
+/*!
+ * \file isl_str_manipulation.h
+ *
+ * \Implementations of isl string manipulator functions
+ *
+ * \date Started: 2016-03-11
+ *
+ * \authors Michelle Strout, Mahdi Soltan Mohammadi
+ *
+ * Copyright (c) 2016, University of Arizona <br>
+ * All rights reserved. 
+ * See ../../COPYING for details. <br>
+ */
+
+#include "isl_str_manipulation.h"
+
+namespace iegenlib{
+
+
+//****** ISL tuple correction functions
+
+/*! This function extracts tuple variables from a Tuple Declaration string.
+**  Tuple Declaration can be either of Relation or Set form
+**  Input:  "[i1,col_tv3_] -> [i3,i4]" or
+**          "[i1,col_tv3_,i3,i4]"  
+**  Output: (pointer to)
+**          {"i1", "col_tv3_", "i3", "i4"}
+*/
+std::string* tupVarsExtract(std::string tupDecl, int inArity, int outArity){
+
+    int st=0, end=0, idxSt = 0;
+    int nparts=1;
+    int arity = inArity + outArity;
+    std::string* tupVars = new std::string[arity];
+
+    if( outArity != 0 ){
+      nparts = 2;
+      arity = inArity;
+    }
+    for(int p=0 ; p < nparts ; p++){
+        st = tupDecl.find_first_of('[', st);
+        end = tupDecl.find_first_of(']', st);
+        std::string str = tupDecl.substr(st+1,end-st-1);
+
+        int tst = 0,tend = 0 ;
+        for(int i = 0 ; i < arity ; i++ ){
+
+            tend = str.find_first_of(',', tst);
+            tupVars[idxSt + i] = str.substr(tst,tend-tst);
+            tst = tend + 1;
+        }
+
+        arity = outArity;
+        idxSt = inArity;
+        st = end + 1;
+    }
+
+    return tupVars;
+}
+/*! This function checks to see whether two Tuple Variables are equal
+**  regardless of spacing before and after their names:
+**  "  ip " == "ip"
+*/
+#define unsigned int uint
+bool tvEqCheck(std::string tv1, std::string tv2){
+    int nonSpSt = 0, nonSpFi = tv1.length()-1;
+    for( uint i = 0; tv1[nonSpSt] == ' ' && i < tv1.length(); i++, nonSpSt++ );
+    for( uint i = 0; tv1[nonSpFi] == ' ' && i < tv1.length(); i++, nonSpFi-- );
```

Comment:
Look at replacing some of this low level code with some kind of trim, see http://stackoverflow.com/questions/216823/whats-the-best-way-to-trim-stdstring.